plugin/core/sessions.py

```diff
@@ -1317,6 +1320,12 @@ def m_workspace_applyEdit(self, params: Any, request_id: Any) -> None:
         self._apply_workspace_edit_async(params.get('edit', {})).then(
             lambda _: self.send_response(Response(request_id, {"applied": True})))
 
+    def m_workspace_codeLens_refresh(self, _: Any, request_id: Any) -> None:
+        """handles the workspace/codeLens/refresh request"""
+        for sv in self.session_views_async():
+            sv.start_code_lenses_async()
```

Comment:
@rwols is it safe to call `start_code_lenses_async` each time ?

Reply:
Also any idea how to fix `error: "SessionViewProtocol" has no attribute "start_code_lenses_async"` ?

Reply:
Should be fine to call it each time. This already happens on editing the document.

Though you should guard the calls with `session.uses_plugin()` like the existing places do:

https://github.com/sublimelsp/LSP/blob/02d0f324142022b247bf4c6a0c3d27eaec7bc364/plugin/documents.py#L547-L547

The type issue can be fixed by declaring `start_code_lenses_async` method on the `SessionViewProtocol` interface.